definitions/npm/styled-components_v5.x.x/flow_v0.104.x-/styled-components_v5.x.x.js

```diff
@@ -248,7 +248,10 @@ declare module 'styled-components' {
   |};
 
   declare export type StyledShorthandFactory<V> = {|
-    [[call]]: <StyleProps, Theme>(
+    [[call]]: (
+      string[]
+    ) => StyledComponent<{ ... }, { ... }, V>,
```

Comment:
Hi! Thanks for helping with this!

At first glance it seems okay. I would maybe put the more specific overload first, and this one after it.

Reply:
Also, I think you could maybe add a special interpolation type that excludes functions, so people can still use interpolation inside the templates. Something like this:

~~~ts
    [[call]]: (
      string[],
      InterpolationBase[]
    ) => StyledComponent<{ ... }, { ... }, V>,
~~~

but from the top of my head, I am not sure if that will cause issues with inference, though

Reply:
@omninonsense sorry for the delay, I've tried to pick this up again but am getting a bit lost

> I would maybe put the more specific overload first

if I put lines 251-253 lower down then the overload currently on line 254 is matched first and flow complains about "Missing type annotation for Theme, StyleProps:.

> a special interpolation type that excludes functions

I think you meant to add a `...`in? Like this?

```
    [[call]]: (
      string[],
      ...InterpolationBase[]
    ) => StyledComponent<{ ... }, { ... }, V>,
```

For some reason I've tried various combinations of the above but am still getting an error 

```
Missing type annotation for V. V is a type parameter declared in function type [1] and was implicitly instantiated at
$ObjMap [2].

     417│     ElementName
     418│   >;
     419│
 [2] 420│   declare type ConvenientShorthands = $ObjMap<
     421│     BuiltinElementInstances,
 [1] 422│     <V>(V) => StyledShorthandFactory<V>
     423│   >;
     424│
     425│   declare interface Styled {
     426│     <Comp: React$ComponentType<P>, Theme, OwnProps = React$ElementConfig<Comp>>(
```

maybe you have time to check the branch out or for a quick call?

Reply:
Hopefully this is recreating the error I'm seeing, will try to debug now:

https://flow.org/try/#0PTAEAEDMBsHsHcBQBLAtgB1gJwC6hwJ7oCmoA3gHKwAmxAvqJFrKqAORbECGAxjmwG4UGbHgBK3Po2asOk-kMSESoAMqFoxaqAC85OouWkAYrFgAeACoA+XaAAkAUQAevHObKJQ30ACMuWABcoFS0iHTWhkSk6gSa1ADCLJgAdsQp7gAKzOgAzgA0oJYAFsSoxLZ6Em72AIK+uThYbkkiaRnm2bB5hWQRiErRoC5NbqrFosVcKdp65gBq1gAU8wCUurax8eOT09TGbtgEC5EDRqAAQgCuyNA4yCmOmuUZAJIpjdM8xLl2nj6gajIABuwWqfCcz3SOAkkHMbCBwLY1nCZyGSRSwPSyGhO1wUxmvz09gA8r4AFYAWS46HMXh811u90eULeHxwXx++Xp3hGzT4eJwBOo9NOaJUWy0guFBz4RxOdiW6x0tlCxCEAMQPFg7NAjTiWmCku0ADJQBisSkcRlpXsieREABIRHBJUbUDmRGgYAougDYjOTC4UDa3WUggAERBdn18QAdIiNT5EEA

Reply:
Hi! On line 20 of the try-flow link, you forgot to call it (also tweaked the function signature to resemble a template literal, so flow doesn't complain about that then):

https://flow.org/try/#0PTAEAEDMBsHsHcBQBLAtgB1gJwC6gEoCmAhgMZ6RayqgDkWJ5tA3IojgJ7qGgDKn0QgBNeAC2w5RxAHZCAYmRzYOAHgBqAPlABeUAAoAzjizJpAcwDaAXQA0oAHSOZHawEodW5607dQAUQAPY0UxCSlZHVB1DT01d20tfg5BEXFccPlFZWjWdi4eACEAV2RoHFM-QVRCaRwASWkjGVJCA0iAb0RQbtAhZAA3AC4CRhwAEkrCatqiSBVaPv7aDUQAXzYfHgBhWGl+muQanFD0mSE23TGAeQAjACsAWWJ0FS6e4tLy6Unp+sacZqtGxvbqBYLkE6SM5vDS5Ui7IygIzJYTDHZ7A5HSEZC6gToASEWwz08S0KkWoGAK3WiEIAUwuFA8P+oAeHAAIgNIsiUvZFgADfmsIA

Reply:
The error is still strange though, lmao.

> I think you meant to add a `...`in? Like this?

Yeah, I forgot the spread

Reply:
> I think you meant to add a `...`in? Like this?

Yeah, I forgot the spread

Reply:
> On line 20 of the try-flow link, you forgot to call it

Cheers, though this solves the issue on this simplified example I still have the same error in my branch. As you say, the error seems strange. I don't think this is intended so I've raised a bug against Flow.

https://github.com/facebook/flow/issues/8344